### ISSUE TITLE
Move PI profile images to official metadata schema

### ISSUE THREAD

**Issue by jeffbaumes** [microbiomedata/nmdc-schema#19]
This is currently in the portal repo. Should we consider something like gravatar? Unsure if ORCID provides public access to a profile image but that might be ideal.

**Comment by wdduncan** [comment]
@dehays yes, that is what I was thinking.

**Comment by dwinston** [comment]
So the plan is to
1. eliminate core/person_value class
2. give core/person class an orcid slot
3. give core/person class an image_url slot
4. remove nmdc/study class' principal_investigator_name slot
5. add nmdc/study class principal_investigator slot with range core/person

Is that right?

**Comment by jeffbaumes** [comment]
@wdduncan I added you as an assignee since I don't think @jbeezley can make the actual query changes himself. Please correct this or delegate if I'm off here.

**Comment by ssarrafan** [comment]
Adding comments from email exchange for reference:

Agree with your comments here David (and yours Kjiersten). I commented in parallel on GitHub, but the gist is that we need to resolve which fields we can and can't expect to require. Many "required" things for the portal could be made optional if needed.

https://github.com/microbiomedata/nmdc-schema/issues/41


On Wed, Apr 28, 2021 at 1:31 PM Kjiersten Fagnan <kmfagnan@lbl.gov> wrote:
I support the approach David laid out for what fields are required vs optional. I can add the following comments to the ticket, but we seem to be getting into this via email. 

Could we create some default values for the portal to populate the page - avatar, URL, DOI and scientific objectives would be harder if not impossible. 

In the future, when contributors are providing data to NMDC, could we also collect - photo, website URL, etc as part of the submission process - or perhaps give the PI the ability to add this themselves? This depends on having some level of access controls (different roles in the data portal than exist right now). Maybe this is part of working directly with the PIs to get their help on the study/data landing pages?

Kjiersten

On Wed, Apr 28, 2021 at 10:20 AM David Hays <dehays@lbl.gov> wrote:
Bill and Emiley said:

- https://github.com/microbiomedata/nmdc-schema/issues/41 I do not have access to the data for some of the fields that Jeff is requesting. I can give an estimate until we track the data.
> @Emiley Eloe-Fadrosh any idea of who to follow up on with to get access to the data needed?  

- https://github.com/microbiomedata/nmdc-schema/issues/19 Again, I do not have access to images of PIs.
> @Emiley Eloe-Fadrosh any idea of who to follow up on with to get access to the PI images needed?

David should be able to address #41, seems like a GOLD database dump issue?

For #19, as was indicated in the github ticket, these were all manually collected by me. Not sure the best solution here, but this could tie into the more general discussion of the study pages (and some items from #41 like scientific objectives that are not part of the GOLD db dump). Not everything can be fully automated.... just my two cents.

_______

On #41, I believe the fields that Bill is referring to are the ones that are NOT available from GOLD; i.e. those listed in https://github.com/microbiomedata/nmdc-metadata/issues/301 and  https://github.com/microbiomedata/nmdc-schema/issues/19 such as PI web site, PI image, scientific objective, publication DOIs.  Basically, the ones that Emiley collected manually and provided to Kitware.

For these, Bill could add these to the schema as non required attributes.  He has no way of making the GOLD ETL populate these because they do not exist in GOLD.   

Jon states that the portal UI depends on these fields - but I believe the portal UI will need to treat them as optional fields as well because normally they will not be available.  If we add 10K studies tomorrow from GOLD or NCBI - we will not be waiting for Emiley to collect values for these fields before they can be displayed in the UI.  The portal UI needs to be flexible enough to handle cases where these values are not available.

I also believe it should not be the responsibility of search portal development to merge additional metadata for studies to extend what was made available for ingest.  So that implies the need for an curate/annotate procedure that is available between GOLD or NCBI ETL and search portal ingest.  And in the case of images - in addition to a way to edit the study json docs to add PI image URLs, there is also the need to add and manage the image files to a location associated with the metadata URL.

So for #41, there are a number of fields that are always available (We will always have a PI for a study.) that can be made required in the schema.  But for those for which there is no available source except manual curation, at best these could be optional fields in the schema. Make sense?

-David

On Tue, Apr 27, 2021 at 1:51 PM Emiley Eloe-Fadrosh <eaeloefadrosh@lbl.gov> wrote:
For these two: 
- https://github.com/microbiomedata/nmdc-schema/issues/41 I do not have access to the data for some of the fields that Jeff is requesting. I can give an estimate until we track the data.
> @Emiley Eloe-Fadrosh any idea of who to follow up on with to get access to the data needed?  

- https://github.com/microbiomedata/nmdc-schema/issues/19 Again, I do not have access to images of PIs.
> @Emiley Eloe-Fadrosh any idea of who to follow up on with to get access to the PI images needed? 

David should be able to address #41, seems like a GOLD database dump issue? 

For #19, as was indicated in the github ticket, these were all manually collected by me. Not sure the best solution here, but this could tie into the more general discussion of the study pages (and some items from #41 like scientific objectives that are not part of the GOLD db dump). Not everything can be fully automated.... just my two cents.

**Comment by wdduncan** [comment]
Please move to May sprint.

**Comment by dwinston** [comment]
@jeffbaumes is this subsumed by / a component of #41?

**Comment by emiley** [comment]
FYI - I think I’m possibly on this thread in error. Perhaps someone
inadvertently at mentioned me rather than the correct recipient.

**Comment by dwinston** [comment]
sorry @emiley, thank you for alerting us! We meant @emileyfadrosh. You can unsubscribe yourself.

https://user-images.githubusercontent.com/1497854/117219195-c2ed4880-add2-11eb-8f39-019ffc9e3fc6.mp4

**Comment by jeffbaumes** [comment]
> @jeffbaumes is this subsumed by / a component of #41?

@dwinston This issue has a slight additional complication attached (we need to host the profile images elsewhere and link to them by URL in the schema) so I feel it could be deserving of its own issue. But I'm also ok rolling it into #41.

**Comment by ssarrafan** [comment]
Based on the meeting today with @dehays, @emileyfadrosh, @dwinston, @wdduncan, and @jbeezley, @wdduncan will add image URL on the person object to the schema.  The images can be stored on an object store at NERSC.

**Comment by ssarrafan** [comment]
Removing Jon from assignee.

**Comment by wdduncan** [comment]
I've added a `profile image url` slot (see PR https://github.com/microbiomedata/nmdc-schema/pull/68).

This is to be use with `study` objects like so:

```json
 {
    "id": "gold:Gs0112340",
    "name": "Thawing permafrost microbial communities from the Arctic, studying carbon transformations",
    "description": "....",
    "principal_investigator": {
        "has_raw_value": "Virginia Rich",
         "profile image url": "http://....." <--- new slot
     }
}
```
**NB**: the property `principal_investigator_name` is **now** named `principal_investigator`.

Closing this ticket. But it can be reopened if needed.
